### PR TITLE
[BEAM-355] Update to Bigtable version 0.3.0

### PR DESCRIPTION
**Pull request by ianzhou1**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

### CODE REVIEW

**Comment by ianzhou1** [comment]
ptal @dhalperi

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java [hunk 1]
```diff
@@ -1019,4 +1026,47 @@ private static String getUserAgent() {
         javaVersion,
         "0.2.3" /* TODO get Bigtable client version directly from jar. */);
   }
+
+  /**
+   * A helper function to produce a Cloud Bigtable user agent string.
+   */
+  public static BigtableOptions.Builder addRetryOptions(BigtableOptions.Builder builder) {
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: Do these need to be public? If not, they should be package-protected.

Also, add tests for these.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java [hunk 2]
```diff
@@ -1019,4 +1026,47 @@ private static String getUserAgent() {
         javaVersion,
         "0.2.3" /* TODO get Bigtable client version directly from jar. */);
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: update

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java (hunk 1 above)
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: Did you file the bug against cloud-bigtable-client to add cloning or toBuilder to these methods?

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableWriteIT.java [hunk 3]
```diff
@@ -83,6 +84,8 @@ public void setup() throws Exception {
         .setClusterId(options.getClusterId())
         .setZoneId(options.getZoneId())
         .setUserAgent("apache-beam-test");
+    bigtableOptionsBuilder = BigtableIO.addBulkOptions(bigtableOptionsBuilder);
+    bigtableOptionsBuilder = BigtableIO.addRetryOptions(bigtableOptionsBuilder);
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableWriteIT.java]: these changes should not be necessary -- that implies that our users will have to include them in their code.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java [hunk 4]
```diff
@@ -388,6 +394,7 @@ public Write withBigtableOptions(BigtableOptions.Builder optionsBuilder) {
       checkNotNull(optionsBuilder, "optionsBuilder");
       // TODO: is there a better way to clone a Builder? Want it to be immune from user changes.
       BigtableOptions.Builder clonedBuilder = optionsBuilder.build().toBuilder();
+      clonedBuilder = addBulkOptions(clonedBuilder);
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: suspect you need the `addRetryOptions` here as well, in some cases.

**Comment by dhalperi** [comment]
R: @dhalperi

Ian, looks like there was some persistent problem in the service because it could not find part of the Google API clients.:

```
java.lang.NoClassDefFoundError: com/google/api/client/util/BackOff
```

Let's debug tomorrow.

**Comment by nevillelyh** [comment]
FYI [bigtable-client 0.9.0](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/releases/tag/bigtable-client-0.9.0) was just released and we verified that they work with our production jobs.

**Comment by dhalperi** [comment]
@nevillelyh yep, thanks. Unfortunately, Bigtable team made backwards-incompatible changes in this release so we'll upgrade to 0.9 separately.

**Pull request closed without merging**